src/app/shared/dialogs/user-profile-dialog/user-profile-dialog.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { SayThanksDialogComponent } from '../say-thanks-dialog/say-thanks-dialog.component';
+
+import { RequestDataService } from './../../services/request-data.service';
+import { AuthUserService } from '../../services/auth-user.service';
+
+import { AchievementsGroup } from '../../models/achievements-group';
+import { Achievement } from './../../models/achievement';
+import { AvatarSize } from './../../enums/avatar-size.enum';
+import { User } from '../../models/user';
+
+@Component({
+  selector: 'app-user-profile-dialog',
+  templateUrl: './user-profile-dialog.component.html',
+  styleUrls: ['./user-profile-dialog.component.scss', '../request-dialog/request-dialog.component.scss']
+})
+export class UserProfileDialogComponent implements OnInit {
+  avatarSize = AvatarSize;
+  isMe$: Observable<boolean>;
+  achievements$: Observable<AchievementsGroup[]>;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public user: User,
+    private matDialog: MatDialog,
+    private authUserService: AuthUserService,
+    private requestDataService: RequestDataService
+  ) { }
+
+  ngOnInit(): void {
+    this.isAuthUserCheck();
+    this.setUserAchievements();
+  }
+
+  setUserAchievements(): void {
+    this.achievements$ = this.requestDataService.getUserAchievement(this.user.id)
+      .pipe(map(this.createAchievementsGroup));
+  }
+
+  createAchievementsGroup(avhievements: Achievement[]): AchievementsGroup[] {
```

Comment:
I would recommend to name all functions with a return type (not void) starting from get. Here 'getGroupedAchievements' would be much better